flake.nix

```diff
@@ -130,64 +131,51 @@
+        haskellModules = perSystem haskellModules;
+
+        # Instruction for the Hercules CI to build on x86_64-linux only, to avoid errors about systems without agents.
+        herculesCI.ciSystems = [ "x86_64-linux" ];
+      };
+      systems = defaultSystems;
+      perSystem = { system, config, pkgs, ... }: {
+        packages = flake.${system}.packages
+          // {
+          default = config.packages."plutip-core:lib:plutip-core";
+        };
 
-      defaultPackage = perSystem (system:
-        let lib = "plutip-core:lib:plutip-core";
-        in self.flake.${system}.packages.${lib});
-
-      packages = perSystem (system: self.flake.${system}.packages);
-
-      apps = perSystem (system: self.flake.${system}.apps);
-
-      devShell = perSystem (system: self.flake.${system}.devShell);
-
-      # This will build all of the project's executables and the tests
-      check = perSystem
-        (system:
-          (nixpkgsFor system).runCommand "combined-check"
-            {
-              nativeBuildInputs = builtins.attrValues self.checks.${system}
-                ++ builtins.attrValues self.flake.${system}.packages;
-            } ''mkdir $out''
-        );
-
-      checks = perSystem (system:
-        self.flake.${system}.checks // {
-          formatting = (nixpkgsFor system).runCommand "formatting-check"
-            {
-              nativeBuildInputs = [
-                self.devShell.${system}.inputDerivation
-                self.devShell.${system}.nativeBuildInputs
-              ];
-            }
-            ''
-              cd ${self}
-              export LC_CTYPE=C.UTF-8
-              export LC_ALL=C.UTF-8
-              export LANG=C.UTF-8
-              export IN_NIX_SHELL='pure'
-              # this check is temporarily skipped in CI due to a bug in
-              # fourmolu:
-              #
-              # ```
-              # Formatting is not idempotent:
-              #   src/Plutip/Launch/Cluster.hs<rendered>:753:19
-              #   before: "       sgs\n         "
-              #   after:  "       sgs{ Ledger.s"
-              # Please, consider reporting the bug.
-              # ```
-              # make format_check cabalfmt_check nixpkgsfmt_check lint
-              mkdir $out
-            '';
-        });
-
-      haskellModules = perSystem haskellModules;
-
-      # Instruction for the Hercules CI to build on x86_64-linux only, to avoid errors about systems without agents.
-      herculesCI.ciSystems = [ "x86_64-linux" ];
+        apps = flake.${system}.apps;
+
+        devShells = {
+          # Adds pre-commit packages and shell hook to the haskell shell
+          default = pkgs.mkShell {
+            inputsFrom = [ config.devShells.haskell config.devShells.dev-pre-commit ];
+            shellHook = config.devShells.haskell.shellHook + config.devShells.dev-pre-commit.shellHook;
+          };
+          haskell = flake.${system}.devShell;
+        };
+
+        checks = flake.${system}.checks;
```

Comment:
You'll just define these in `perSystem` so you main flake record will be empty most likely